src/main/java/edu/kit/compiler/optimizations/Util.java

```diff
@@ -236,4 +238,36 @@ public void visitBlock(Block block) {
 
     }
 
+    /**
+     * Call `func` for each non-Bad direct control flow predecessor of `block`.
+     */
+    public static void forEachPredBlock(Block block, ObjIntConsumer<Block> func) {
```

Comment:
`CommonUtil.stream(block.getPreds()).filter(..).forEach(..)`?

Reply:
Unfortunately, I usually need both the predecessor `Block`, as well as the index of the predecessor.

Reply:
Unfortunately, Java has no tuples :D

Reply:
Not shorter or easier to read, but I'll just leave this here:

```java
public static void forEachPredBlockStream(Block block, ObjIntConsumer<Block> func) {
    IntStream.range(0, block.getPredCount())
        .mapToObj(i -> new AbstractMap.SimpleEntry<>(i, block.getPred(i)))
        .filter(item -> item.getValue().getOpCode() != ir_opcode.iro_Bad)
        .forEach(item -> func.accept((Block) item.getValue().getBlock(), item.getKey()));
}
```